vid/src/advz/test.rs

```diff
@@ -261,6 +261,43 @@ fn verify_share_with_multiplicity() {
     }
 }
 
+#[test]
+fn sad_path_verify_share_with_multiplicity() {
+    let advz_params = AdvzParams {
+        recovery_threshold: 16,
+        num_storage_nodes: 20,
+        multiplicity: 32, // payload fitting into a single polynomial
+        payload_len: 8200,
+    };
+    let (mut advz, payload) = advz_init_with(advz_params);
+
+    let disperse = advz.disperse(payload).unwrap();
+    let (shares, common, commit) = (disperse.shares, disperse.common, disperse.commit);
+    for (i, share) in shares.iter().enumerate() {
+        // corrupt the last evaluation of the share
+        {
+            let mut share_bad_eval = share.clone();
+            share_bad_eval.evals[common.multiplicity as usize - 1].double_in_place();
+            advz.verify_share(&share_bad_eval, &common, &commit)
+                .unwrap()
+                .expect_err("bad share value should fail verification");
+        }
+
+        // corrupt the first eval proof of this share by assigning it the value of last
```

Comment:
I thought the purpose of this test is to corrupt evals (and their proofs) for shares *beyond* the first, no?

Also, maybe add a comment for this test explaining that it's a regression test for #654 .

Reply:
> I thought the purpose of this test is to corrupt evals (and their proofs) for shares _beyond_ the first, no?

Yes, that is better. Modified here: [1fd8293](https://github.com/EspressoSystems/jellyfish/pull/657/commits/1fd8293edba68a4b7c8ea943238c5acc2b193fd7)

Comment added here: [049d20d](https://github.com/EspressoSystems/jellyfish/pull/657/commits/049d20dae4fecb698122cff6ff08db653b2d2da6)